packages/preact-router/src/CatchBoundary.tsx

```
@@ -0,0 +1,121 @@
+  getResetKey: () => number | string
+  children: ComponentChildren
+  errorComponent?: ErrorRouteComponent
+  onCatch?: (error: Error, errorInfo: { componentStack?: string }) => void
+}) {
+  const errorComponent = props.errorComponent ?? ErrorComponent
+
+  return (
+    <CatchBoundaryImpl
+      getResetKey={props.getResetKey}
+      onCatch={props.onCatch}
+      children={({ error, reset }) => {
```

Biome (2.1.2)

[error] Avoid passing children using a prop

The canonical way to pass children in React is to use JSX elements

(lint/correctness/noChildrenProp)